### DISCUSSION TITLE
zsh "maximum nested function level reached" during startup when shell-integration is enabled

### DISCUSSION THREAD

**Discussion by anund** [ghostty-org/ghostty#3982] [Feature Requests, Ideas]
From https://github.com/ghostty-org/ghostty/issues/3453#issuecomment-2564941206. Use this discussion for debugging.

**Comment by anund** [comment]
@devashishraj Your linked config https://gist.github.com/devashishraj/3354a90c99bbdd725f3673261ea8934d references several things on the local disk. Can you go through a cycle of commenting things out until the bug goes away?
I can see if just powerline + ohmyzsh reproduces the issue in parallel.

**Reply by anund** [reply]
Running a vm with just oh-my-zsh and a ~default p10k config does not seem to generate any issues. Perhaps one of the other setups in your .zshrc is the problem? If you can narrow done the lines or set of lines that results in the init loop that would be helpful.

**Reply by devashishraj** [reply]
> Running a vm with just oh-my-zsh and a ~default p10k config does not seem to generate any issues. Perhaps one of the other setups in your .zshrc is the problem? If you can narrow done the lines or set of lines that results in the init loop that would be helpful.

oh, i will check what section of init, this narrow down to.

**Reply by devashishraj** [reply]
@anund i removed the `fast-syntax-highlighting` as per this [comment](https://github.com/ohmyzsh/ohmyzsh/issues/12869#issuecomment-2572842608) and it works .

**Reply by anund** [reply]
Well we have some progress. :partying_face: Poking around with a [nixos vm](https://gist.github.com/anund/3651fb29db9922d6a5e1f751318c11c1) and just adding powerline + ohmysh + ohmyzshplugins listed doesn't reproduce the behaviour. It's possible I haven't reproduced the `.zshrc` equivalent so I included the currently generate output for reference in the gist.

Two follow up questions: 
- If you strip everything out of your `.zshrc` beside powerline/ohmyzsh/ohmyzshplugins does the issue still happen?
- What version/git hash are each of the installed plugins?

**Reply by devashishraj** [reply]
> Well we have some progress. 🥳 Poking around with a [nixos vm](https://gist.github.com/anund/3651fb29db9922d6a5e1f751318c11c1) and just adding powerline + ohmysh + ohmyzshplugins listed doesn't reproduce the behaviour. It's possible I haven't reproduced the `.zshrc` equivalent so I included the currently generate output for reference in the gist.
> 
> Two follow up questions:
> 
> * If you strip everything out of your `.zshrc` beside powerline/ohmyzsh/ohmyzshplugins does the issue still happen?
> * What version/git hash are each of the installed plugins?


a) i did strip out everything but and issue still persisted unless i removed that `fast...` plugin , i will look into zshrc throughly on weekend.
b ) version is :
```
omz version
master (d826691)
```
as for specific versions of plugins i don't saw any info like that.

**Reply by anund** [reply]
hmm, it looks like `zsh-syntax-highlighting` `fast-syntax-highlighting` `zsh-autocomplete` all need manual installation. They should be git repositories and have an associated git hash.

**Reply by anund** [reply]
I did some fishing with the `FUNCNEST` value today. I could set the value as low as ~25 without issue. Oh-my-zsh version from ~[October](https://github.com/NixOS/nixpkgs/blob/76af1d7fed03d93fe9d509be57573dc22d713249/pkgs/by-name/oh/oh-my-zsh/package.nix#L29) last year. I also updated to the oh-my-zh version you mention. I didn't see a difference. (updated the gist)

**Comment by mitchellh** [comment]
This is a dupe.

**Reply by anund** [reply]
Er yes? The associated issue is not currently actionable.

**Reply by mitchellh** [reply]
Ah my mistake. Thanks

**Reply by anund** [reply]
Nw.
